internal/pkg/otel/diagnostics.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package otel
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/otel/extension/elasticdiagnostics"
+	"github.com/elastic/elastic-agent/pkg/control/v2/client"
+)
+
+func PerformDiagnosticsExt(ctx context.Context, cpu bool) (*elasticdiagnostics.Response, error) {
```

Comment:
```suggestion
func PerformDiagnosticsExt(ctx context.Context, includeCPUProfile bool) (*elasticdiagnostics.Response, error) {
```

I like the parameter name to be more descriptive. Can you also add a docstring to this function?

Reply:
DOne